hazelcast/src/main/java/com/hazelcast/internal/ascii/memcache/SetCommand.java

```diff
@@ -59,15 +60,11 @@ public boolean readFrom(ByteBuffer src) {
         return false;
     }
 
-    void copy(ByteBuffer cb) {
-        if (cb.isDirect()) {
-            int n = Math.min(cb.remaining(), bbValue.remaining());
-            if (n > 0) {
-                cb.get(bbValue.array(), bbValue.position(), n);
-                upcast(bbValue).position(bbValue.position() + n);
-            }
+    void copy(ByteBuffer src) {
+        if (src.isDirect()) {
```

Comment:
The `bbValue` is always on the heap, if I read it correctly. So wouldn't it be simpler to use just the `copyToHeapBuffer` without checking the `src` type here?

Reply:
Makes sense, I have addressed this and other comments in https://github.com/hazelcast/hazelcast/pull/21624/commits/c012f44cd1baee7d05fe48d18d48ae6e7a631fea